client/app/components/Select.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { useState, useEffect } from "react";
+import { maxBy } from "lodash";
+import AntdSelect, { SelectProps } from "antd/lib/select";
+import { calculateTextWidth } from "@/lib/calculateTextWidth";
+
+function Select({ options, ...props }: SelectProps<any>): JSX.Element {
```

Comment:
I also don't have much better ideas to name this component 😬 , maybe `SelectWithVirtualScroll`?